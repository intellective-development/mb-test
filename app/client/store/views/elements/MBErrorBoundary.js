// @flow

import * as React from 'react';

// note that this inexact type allows us to pass as much info as we'd like
type ExtraErrorData = {
  location: string
};

type MBErrorBoundaryProps = {
  children: React.Node,
  errorData: (error: Object, info: Object) => ExtraErrorData
}
type MBErrorBoundaryState = {
  has_error: boolean
};

export class MBErrorBoundary extends React.Component<MBErrorBoundaryProps, MBErrorBoundaryState> {
  state = { has_error: false }

  componentDidCatch(error: Error, info: Object){
    this.setState({ has_error: true });

    // sanity check, errorData really should always be set
    const extra_data = this.props.errorData ? this.props.errorData(error, info) : {};
    Raven.captureException(error, {
      extra: { react_info: info, ...extra_data}
    });
  }

  render(){
    if (this.state.has_error) return null;

    return this.props.children;
  }
}

export default MBErrorBoundary;
