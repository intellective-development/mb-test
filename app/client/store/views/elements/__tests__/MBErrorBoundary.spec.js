import * as React from 'react';
import MBErrorBoundary from '../MBErrorBoundary';

describe('MBErrorBoundary', () => {
  it('renders', () => {
    expect(render(
      <MBErrorBoundary errorMessage={() => 'Some message'}>
        <div>A thing to click outside of</div>
      </MBErrorBoundary>
    )).toMatchSnapshot();
  });
});
