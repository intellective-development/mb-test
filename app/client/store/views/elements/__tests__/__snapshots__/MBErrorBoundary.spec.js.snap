// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MBErrorBoundary renders 1`] = `
<div>
  A thing to click outside of
</div>
`;
